fee_allocator/accounting/interfaces.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABC
+
+"""
+`CorePoolChain` and `PoolFee` reference each other so they need to be abstract to prevent circular imports.
```

Comment:
Wonder if there is a way we can do this without circural dependancies and needing to get into abstract classes. I think I understand why this is happening, but it's confusing and hard to think about and leads to more jumping around in the code when trying to debug.

Would be really good to come up with an architecture diagram or something and then think about the structure.  Maybe I'llt ry to get a decent UML output.

Reply:
we do this so that:
- core pools are aware of chain data
- chains are aware of core pool data

which is needed for doing a lot of the fee split calculations